src/messages/MOSDRepOp.h

```diff
@@ -83,9 +119,15 @@ class MOSDRepOp : public Message {
     ::decode(from, p);
     ::decode(updated_hit_set_history, p);
     ::decode(pg_trim_rollback_to, p);
+    final_decode_needed = false;
   }
 
   virtual void encode_payload(uint64_t features) {
+    if ((features & CEPH_FEATURE_NEW_OSDREPOP_ENCODING) == 0) {
+      header.version = 1;
+    } else {
+      header.version = HEAD_VERSION;
```

Comment:
in theory we should set compat version here too, although since the peer has the feature it won't matter